src/client/LambdasClient.ts

```diff
@@ -1,190 +1,36 @@
-  ): Promise<OwnedItems<B>>
-  fetchCatalystsApprovedByDAO(options?: RequestOptions): Promise<ServerMetadata[]>
-  fetchLambdasStatus(options?: RequestOptions): Promise<{ contentServerUrl: string }>
-}
-
-export function createLambdasClient(options: ClientOptions): LambdasClient {
+export function createLambdasClient(options: ClientOptions) {
   const lambdasUrl = sanitizeUrl(options.url)
   const fetcher = withDefaultHeadersInjection(options.fetcher)
 
-  async function fetchProfiles(ethAddresses: string[], options?: RequestOptions): Promise<any[]> {
-    if (ethAddresses.length === 0) {
-      return Promise.resolve([])
-    }
-
-    const requestOptions = mergeRequestOptions(options ? options : {}, {
-      body: JSON.stringify({ ids: ethAddresses }),
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' }
-    })
-
-    return (await fetcher.fetch(`${lambdasUrl}/profiles`, requestOptions)).json()
-  }
-
-  function fetchWearables(filters: WearablesFilters, options?: RequestOptions): Promise<any[]> {
-    const queryParams = convertFiltersToQueryParams(filters)
-    if (queryParams.size === 0) {
-      throw new Error('You must set at least one filter')
-    }
-
-    return splitAndFetchPaginated({
-      fetcher: fetcher,
-      options,
-      baseUrl: lambdasUrl,
-      path: '/collections/wearables',
-      queryParams,
-      uniqueBy: 'id',
-      elementsProperty: 'wearables'
-    })
-  }
-
-  function fetchOwnedWearables<B extends boolean>(
-    ethAddress: string,
-    includeDefinitions: B,
-    options?: RequestOptions
-  ): Promise<OwnedItems<B>> {
-    return splitAndFetch<B extends false ? OwnedItemsWithoutDefinition : OwnedItemsWithDefinition>({
-      baseUrl: lambdasUrl,
-      path: `/collections/wearables-by-owner/${ethAddress}`,
-      queryParams: { name: 'includeDefinitions', values: [`${includeDefinitions}`] },
-      fetcher,
-      options
-    })
-  }
-
-  function fetchOwnedThirdPartyWearables<B extends boolean>(
-    ethAddress: string,
-    thirdPartyId: string,
-    includeDefinitions: B,
-    options?: RequestOptions
-  ): Promise<OwnedItems<B>> {
-    const queryParams = new Map([
-      ['collectionId', [thirdPartyId]],
-      ['includeDefinitions', [`${includeDefinitions}`]]
-    ])
-    return splitAndFetch<B extends false ? OwnedItemsWithoutDefinition : OwnedItemsWithDefinition>({
-      fetcher,
-      options,
-      baseUrl: lambdasUrl,
-      path: `/collections/wearables-by-owner/${ethAddress}`,
-      queryParams
-    })
-  }
-
-  function fetchEmotes(filters: EmotesFilters, options?: RequestOptions): Promise<any[]> {
-    const queryParams = convertFiltersToQueryParams(filters)
-    if (queryParams.size === 0) {
-      throw new Error('You must set at least one filter')
+  function wrap<T extends (...args: any) => CustomClient<any>>(f: T) {
+    return (...args: Parameters<T>): ReturnType<ReturnType<T>> => {
+      return f(args)(lambdasUrl, fetcher) as any
```

Comment:
If the args that do not change are lambdasUrl and fetcher shouldn't this decorator be the other way around, so this module exports:

`
  getLambdaStatus: client.getLambdaStatus(lambdasUrl, fetcher) -> getLambdasStatus expecting args
  getCollections: client.getCollections(lambdasUrl, fetcher) -> getCollections expecting args 
`

Reply:
that would be awesome, but the problem is this is the autogenerated method (one example):

```
/**
 * Returns the list of third party integrations and collections
 * @summary Returns the list of third party integrations as well as collections
 */
export const getThirdPartyIntegrations = () => {
  return useCustomClient<ThirdPartyIntegrations>({ url: `/lambdas/third-party-integrations`, method: 'get' })
}
```

so that receives the request arguments, the only way I found to hook this to our fetcher is to provide a customClient that can be used as a function that gets the fetcher. It's a bit convoluted but we have autogenerated everything so I think its worth the price.